src/solc_0.8/polygon/child/asset/PolygonAssetERC721.sol

```diff
@@ -4,19 +4,31 @@ pragma solidity 0.8.2;
 import {BaseERC721} from "../../../assetERC721/BaseERC721.sol";
 import {IPolygonAssetERC721} from "../../../common/interfaces/IPolygonAssetERC721.sol";
 import {IERC721Base} from "../../../common/interfaces/IERC721Base.sol";
+import {
+    DefaultOperatorFiltererUpgradeable
+} from "../../../OperatorFilterer/contracts/upgradeable/DefaultOperatorFiltererUpgradeable.sol";
+import {
+    OperatorFiltererUpgradeable
+} from "../../../OperatorFilterer/contracts/upgradeable/OperatorFiltererUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 
 /// @title This contract is for AssetERC721 which can be minted by a minter role.
 /// @dev AssetERC721 will be minted only on L2 and can be transferred to L1 but not minted on L1.
 /// @dev This contract supports meta transactions.
 /// @dev This contract is final, don't inherit from it.
-contract PolygonAssetERC721 is BaseERC721, IPolygonAssetERC721 {
+contract PolygonAssetERC721 is BaseERC721, IPolygonAssetERC721, DefaultOperatorFiltererUpgradeable {
     bytes32 public constant METADATA_ROLE = keccak256("METADATA_ROLE");
```

Comment:
Is this upgradeable or should the value be set in the initialiser?

Reply:
This is upgradeable and the value is set in the **initialize** function with __OperatorFilterer_init

Reply:
Though I need to change DefaultOperatorFiltererUpgradeable to OperatorFiltererUpgradeable in the inheritance

Reply:
Sorry, I meant for METADATA_ROLE

Reply:
I think not that hash would be constant so to put it in the source code of the contract would save gas.